src/server/replica.cc

```diff
@@ -231,9 +236,9 @@ void Replica::MainReplicationFb() {
 
     // 3. Initiate full sync
     if ((state_mask_.load() & R_SYNC_OK) == 0) {
-      if (HasDflyMaster())
-        ec = InitiateDflySync();
-      else
+      if (HasDflyMaster()) {
+        ec = InitiateDflySync(std::exchange(data, nullopt));
```

Comment:
why have you use exchange and not std::move?
do you use access `data` afterwards?

Reply:
this is the main loop we can call InitiateDflySync again if we do reconnect